docs: update telemetry documentation (#972)
awslabs/aws-advanced-jdbc-wrapper: docs/using-the-jdbc-driver/Telemetry.md

## Using Telemetry

> Telemetry for the AWS JDBC Driver is a monitoring strategy that overlooks all plugins enabled in [`wrapperPlugins`](./UsingTheJdbcDriver.md#connection-plugin-manager-parameters) and is not a plugin in itself. Therefore no changes are required in the `wrapperPlugins` parameter to enable Telemetry.

Reviewer: ```suggestion
Telemetry for the AWS JDBC Driver is a monitoring strategy that overlooks all plugins enabled in [`wrapperPlugins`](./UsingTheJdbcDriver.md#connection-plugin-manager-parameters) and is not a plugin in itself. No changes are required in the `wrapperPlugins` parameter to enable Telemetry.
```

Applications that interact with the AWS JDBC Driver may or may not have already opened telemetry traces on their own. In this case, it is up to the user to determine how they want to mix both application and driver traces.

This can be done using the AWS JDBC Driver's `telemetrySubmitTopLevel` property. This property allows users to choose to submit the driver traces always as top level traces or to submit them nested to the application traces.

> The default value is set to `false`, which means the driver traces to always be nested into their application traces. That will happen unless there are no open application traces when the driver is running. In that case no driver traces will be collected or submitted. When the property is set to `true`, all the driver traces would be submitted top level traces. If the application has already open traces, it will not be possible for the driver traces to be top level and the driver traces therefore will not be submitted.

Reviewer: ```suggestion
The default value is set to `false`, which means the driver traces will always be nested into the application traces. If there are no open application traces when the driver is running, the driver traces will not be nested into the application traces and no driver traces will be collected or submitted. When the property is set to `true`, all the driver traces will be submitted as top level traces. If the application already has open traces, it will not be possible for the driver traces to be top level, and the driver traces will not be submitted.
```
